### ISSUE TITLE
Rating A11y Pass

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#7096]
Tracking all of the issues brought up with the 11/13 accessibility pass on Rating  control
Details on individual bugs can be found on [SharePoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B46e7845f-6c01-46a3-80be-c474753aad6f%7D&action=default&uid=%7B46E7845F-6C01-46A3-80BE-C474753AAD6F%7D&ListItemId=1536&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble
)

- [x] ~~Bug 1: [Accessibility][Usable]: Screen gets scroll up when user navigates on rating starts in scan mode.~~ No repro as you can see in the screenshot in [this comment](https://github.com/OfficeDev/office-ui-fabric-react/issues/7096#issuecomment-459891683)
- [x] Bug 2: [Accessibility][Visual requirements]: Color contrast ratio of the 'Non-selected rating stars' is less than 4.5:1. PR #8246 
- [x] ~~Bug 3: [Accessibility][Forms and validation]: Label is not associated with rating stars.~~ No repro as you can see in the screenshot in [this comment](https://github.com/OfficeDev/office-ui-fabric-react/issues/7096#issuecomment-459891683)
- [x] Bug 4: [Accessibility][Programmatic access]: Name is not provided for 'Button Controlled Rating' rating stars. PR #8246 
- [x] ~~Bug 5: [Accessibility][Screen reader]: 'Button Controlled Rating' starts are not accessible in scan mode.~~ This is correct behavior for all examples that are "readonly" and "disabled" by definition. You can see that in the regular examples, "Disabled", "Half star in readOnly mode" and "Custom icons" are all not accessible in scan mode, because this respects the "readonly" and "disabled" properties.
- [x] ~~Bug 6: [Accessibility][Programmatic access]: After selecting 'Click to change rating to 0' button narrator does not announce anything.~~ Explained why this is correct behavior in [comment](https://github.com/OfficeDev/office-ui-fabric-react/issues/7096#issuecomment-459891683) - follows w3 reference.
- [x] ~~Bug 7: [Accessibility][Programmatic access]: State of the rating stars is not defined properly.~~ Explained why this is correct behavior in [comment](https://github.com/OfficeDev/office-ui-fabric-react/issues/7096#issuecomment-459891683) - follows w3 reference.

Component: Rating
@micahgodbolt

**Comment by aneeshack4** [comment]
Running into some problems when I try to repro and investigate these bugs - Narrator keeps saying "No item in view" when I try to navigate the Rating control page on the Fabric website.... @TestAccessibilityGIT Did you ever come across this problem too?

**Comment by TestAccessibilityGIT** [comment]
Hi Aneesha,
We haven’t come across this issue. I hope you are using the RS5 latest OS build.

Regards,
Arvind

**Comment by aneeshack4** [comment]
@TestAccessibilityGIT This is the windows version I'm on: 
![image](https://user-images.githubusercontent.com/4161791/50362060-02c17700-051b-11e9-87c6-9e0550d4297d.png) Is this what you meant by RS5?
It's hard to update to a different build for each bug that we across so can you please provide a Codepen that repros these bugs? That will make these bugs actionable for us to fix. Here's one with the Rating Fabric component that you can get started with: https://codepen.io/pen/?&editable=true

**Comment by aneeshack4** [comment]
@TestAccessibilityGIT When I checked for updates, it says my machine just recently updated so I don't think I can do much more on my end to get a repro.
![image](https://user-images.githubusercontent.com/4161791/50362395-6dbf7d80-051c-11e9-8cdd-03c725f455f8.png)

Please use the Rating Codepen I linked in my earlier comment.

**Comment by TestAccessibilityGIT** [comment]
I think we already had discussion on this in past. I’m attaching that thread again how to update OS to latest version.

Regards,
Arvind

**Comment by aneeshack4** [comment]
@TestAccessibilityGIT I looked up which build # corresponds to RS5 and it looks like I'm actually already on RS5:
![image](https://user-images.githubusercontent.com/4161791/50363118-31d9e780-051f-11e9-9c7b-560824199b6f.png)

Here's a [video](https://microsoft-my.sharepoint-df.com/:u:/p/aneeshak/EVxkzLg1cWdGmWr-PuXEiukBPqeASegkAEXlOflBExJ2mQ?e=58xWum) showing this "No item in view" behavior that Narrator's exhibiting when interacting with the Rating component. This actually seems to be a larger problem than the bugs reported in this issue. Maybe if Narrator's actually able to recognize the items in the current view, it will repro the bugs you found.

**Comment by aneeshack4** [comment]
@TestAccessibilityGIT I revisited this & was able to get it to stop doing the "no item in view" nonsense and actually get some repros. I'm able to repro all of the issues except 1 & 3. I didn't notice any scroll up during scan mode & as you can see in the narrator buddy output, the label of the rating stars is clearly stated.
![image](https://user-images.githubusercontent.com/4161791/52153009-3d93f000-262d-11e9-8263-1534f4fba525.png)
Some questions about the rest:
Bug 6 - what _should_ Narrator announce in this case? In this w3 example, nothing is announced when an equivalent button "submit rating" is selected. 
Bug 7 - which rating stars are you referring to here? If you're referring to the rating stars above the button "click to change rating to 5", in the w3 reference example, the state isn't read out either. 

@micahgodbolt, do you think we should be reading out the state of the rating stars after the "click to change rating to 5" button's pressed & announce that the button was pressed? It seems redundant because the button says what's going to happen and the button's label itself is being read out and there's an audio cue signifying that the button's action was implemented. As in the w3 reference example, for Bug 6 and Bug 7, I'm not sure if anything else needs to be announced. 

I'll work on getting a color that meets the contrast ratio to resolve Bug 2. And I'll try and find a fix to Bugs 4 and 5 where the Button Controlled Rating stars are not announced/not accessible in scan mode. These bugs don't repro on the w3 reference example in scan mode so I believe ours need to be fixed in that regard.
